### ISSUE TITLE
Opening report via deep link animates the page twice

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23441]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to a report
2. Copy the url to clipboard
3. Return to LHN
4. Minimise the app
5. Paste the url in an external app like Notes
6. Tap on the report url
## Expected Result: 
NewDot opens and navigates to the corresponding report
## Actual Result: 
App opens and the report page gets animated in twice
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.44-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/23f9d604-d6e1-49c3-888c-3dda8d01e031

https://github.com/Expensify/App/assets/43996225/d72a7517-e866-4ff8-92a7-9e8635571c57

**Expensify/Expensify Issue URL:**
**Issue reported by:** @aswin-s
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690060285079399

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kevinksullivan** [comment]
Unable to produce this as the link takes me to mWeb. Closing unless someone can correct the steps or let me know what's wrong.


https://github.com/Expensify/App/assets/22508290/dc7a6860-3235-4461-a1ac-631326dd3ad1

**Comment by aswin-s** [comment]
@kevinksullivan I can still reproduce this bug. I'm on latest prod build `1.3.42-26`. Also I noticed that the link gets opened in mWeb, which should never happen on staging or prod builds. Deep links should get opened in native app itself.

Steps followed.

1. Navigate to a group chat
2. Copy URL
3. Navigate back to LHN. (Important)
4. Minimise NewDot and paste the URL in notes.
5. Tap on the URL to open the deep in in NewDot. 

https://github.com/Expensify/App/assets/6880914/34c20b15-cbfd-4b1d-9dc6-cd60ccc930b2

**Comment by aswin-s** [comment]
Reproduced on TestFlight version `1.3.44-2` as well.

**Comment by aswin-s** [comment]
@kevinksullivan Shouldn't this be reopened?

**Comment by kbecciv** [comment]
Issue is reproduced on build 1.3.45.3

https://github.com/Expensify/App/assets/93399543/9fd9b3eb-ff92-4156-bedc-6afbb2148a1a

**Comment by kevinksullivan** [comment]
Moving forward

**Comment by abdulrahuman5196** [comment]
Reviewing today

**Comment by abdulrahuman5196** [comment]
Sorry wasn't able review that time. Will give a review in my morning

**Comment by aswin-s** [comment]
I can still reproduce this issue on latest TestFlight build 1.3.50

**Comment by kevinksullivan** [comment]
Hi @sophiepintoraetz I am going OOO for the week so tapping you in the interim. Thanks!

**Comment by abdulrahuman5196** [comment]
On @tienifr 's solution here https://github.com/Expensify/App/issues/23441#issuecomment-1656725230
I tested the solution and it seems to be working fine. 

But I need to confirm on this root cause

>After the navigation refactor, the deep linking already works out of the box, but in here we still keep the legacy logic where we manually handle the navigation to a specific report. That manual handling is not necessary now. This leads to the double navigation and the visual issue.

Tagging @WoLewicki / @mountiny   as well to see if this is true about navigation refractor. Even in the other PR which this is mentioned https://github.com/Expensify/App/issues/20624 also says we are seeing this issue recently.

Meanwhile @tienifr , could you also share more information if you have any on the root cause? And if we are saying manual handling is not required, then `openReportFromDeepLink` function itself is not required, is it?

**Comment by mountiny** [comment]
@abdulrahuman5196 I swear we have tested this before and it was fine so possible it might be a regression from something lately.

@WoLewicki @abdulrahuman5196 any ideas for the proposed fix or where this could be coming from?

**Comment by WoLewicki** [comment]
>  And if we are saying manual handling is not required, then openReportFromDeepLink function itself is not required, is it?

Yeah, we could check if removing this flow destroys any behaviors. I haven't participated in implementing this part of refactor, so I won't say for sure that it will work correctly then.

**Comment by sophiepintoraetz** [comment]
Not overdue - still discussing solutions.

**Comment by abdulrahuman5196** [comment]
I did some testing and have some information.

1) This issue is not reproducible if you are already on chat page and click on deeplink


https://github.com/Expensify/App/assets/46707890/9b44545a-54fc-4593-9fc9-4a77c94bd098



2) Issue is reproducible if you are already on the home page and click on deeplink


https://github.com/Expensify/App/assets/46707890/90b59e7b-918c-4f48-8e86-332378ad9195


3) It should be a regression from this PR - https://github.com/Expensify/App/issues/20624
I tested removing that change PR and I am unable to reproduce this issue. On investigating the videos of the PR, the videos where taking from chat page in which cases this animate twice bug is not seeing. And the video is stop as soon as the page is opened, but on some cases it takes a second for the second page to animate. So it could have been missed in the PR.

**Comment by abdulrahuman5196** [comment]
And @tienifr could you kindly confirm if we will bring this bug https://github.com/Expensify/App/issues/20624 back since we are actually remove the change from there itself.

And we don't have solid rootcause on why this is happening, if possible could you check on the above information and see if we can piece the root cause together?

**Comment by mountiny** [comment]
@tienifr @abdulrahuman5196 can you check if its repro on this branch build? https://github.com/Expensify/App/pull/22437

**Comment by abdulrahuman5196** [comment]
> @tienifr @abdulrahuman5196 can you check if its repro on this branch build? #22437

Are we trying to fix this there? Or something related we should wait on?

**Comment by mountiny** [comment]
@abdulrahuman5196 do but its changing a bit the navigators structure so I though it might have helped

**Comment by kevinksullivan** [comment]
@abdulrahuman5196 can you provide an update here?

**Comment by abdulrahuman5196** [comment]
I am planning to approve @tienifr 's proposal here https://github.com/Expensify/App/issues/23441#issuecomment-1656725230

If there is no plan to fix this bug part of separate issue.

cc: @allroundexperts @mountiny

**Comment by mountiny** [comment]
> If there is no plan to fix this bug part of separate issue or this https://github.com/Expensify/App/issues/20624

Sorry can you please re-phrase this, I am not sure if I understood.

So if we go with @tienifr proposal it seems like it will introduce:

> Issue is reproducible if you are already on the home page and click on deeplink

this could be an issue when you get a loud notification while at the app and you click on it, right? then the deeplink will navigate you twice

Anything else found so far?

**Comment by mountiny** [comment]
What did @abdulrahuman5196 meant by:
> If there is no plan to fix this bug part of separate issue or this https://github.com/Expensify/App/issues/20624

I think we could go with @tienifr proposal then, do you agree @abdulrahuman5196 ?

**Comment by abdulrahuman5196** [comment]
@mountiny Sorry for the confusion. Let me clarify.

There were 2 parallel things going on related to this issue

1) This issue is most likely a regression from here https://github.com/Expensify/App/issues/20624
2) Some navigation structure change you had previously mentioned might fix this https://github.com/Expensify/App/issues/23441#issuecomment-1679355113

As far as the current status, I don't see both the above path leading to fixing this issue.

So my question was, can I proceed by treating/reviewing @tienifr 's solution (https://github.com/Expensify/App/issues/23441#issuecomment-1656725230) to fix this issue, instead of waiting on external paths?
I just have to re-confirm, the fix doesn't bring back this bug - https://github.com/Expensify/App/issues/20624. If that's successful, I am good to approving his proposal from my end.

**Comment by mountiny** [comment]
Perfect, thanks for the write up

**Comment by DanutGavrus** [comment]
@abdulrahuman5196 
> Some navigation structure change you had previously mentioned might fix this

It might, because if I remove my PR change, **no** bug happens anymore(the one I fixed 1 month ago https://github.com/Expensify/App/issues/20624 & this new one https://github.com/Expensify/App/issues/23441). I have not been able to find the changes which fixed my original Issue as I've explained [here](https://github.com/Expensify/App/issues/20624#issuecomment-1686573606), but we [concluded](https://github.com/Expensify/App/issues/20624#issuecomment-1692190368) to treat that as a regression as I could not find changes which could have broken my fix by adding this new issue either, thus it was most likely a regression.

> If there is no plan to fix this bug part of separate issue or this https://github.com/Expensify/App/issues/20624

I've provided a fix for this **new** issue [there](https://github.com/Expensify/App/issues/20624#issuecomment-1660102692). I've tried all 3 approaches: implementing that fix, removing my PR change & completely removing manual report navigation as it's now handled by the navigation itself(@tienifr's Proposal). I confirm that neither the **old** or this **new** bug happens with either of them. My fix just adds complexity, I prefer @tienifr's solution as it's much more elegant. Sibtain(C+ from my PR) also prefers @tienifr's approach as stated [here](https://github.com/Expensify/App/issues/20624#issuecomment-1660830907). So I think there's no plan to fix that there.

**Comment by kevinksullivan** [comment]
Hi @abdulrahuman5196 , just checking in. Will you be able to check out the response above this week?

**Comment by mountiny** [comment]
Bump @abdulrahuman5196

**Comment by abdulrahuman5196** [comment]
Tested @tienifr 's proposal here https://github.com/Expensify/App/issues/23441#issuecomment-1656725230 and it looks good and works well and doesn't cause any regression.

🎀 👀 🎀
C+ Reviewed

cc: @mountiny

**Comment by aswin-s** [comment]
Offer seems to be for 50$ for reporter. It is supposed to be 250$ for issues created in July, right?

**Comment by mountiny** [comment]
ye will be handled once we pay out

**Comment by sophiepintoraetz** [comment]
Whoops - forgot to unsubscribe to this one and paid @aswin-s $250.

**Comment by kevinksullivan** [comment]
Awaiting payment, @abdulrahuman5196 please finish out the checklist above. Thanks!

**Comment by abdulrahuman5196** [comment]
>The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/22740

> The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

https://github.com/Expensify/App/issues/20624#issuecomment-1676360204

>Determine if we should create a regression test for this bug.
If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

No need for new test. Normal deeplink regression tests should be fine.

**Comment by abdulrahuman5196** [comment]
@kevinksullivan Completed the checklist

**Comment by kevinksullivan** [comment]
All set
